### PR TITLE
Add tests and configure GitHub repository

### PR DESCRIPTION
**Pull request by JoelSpeed**
This repo had no tests, so I've added some tests that check the general functionality of the code generation so that we can more easily identify bugs in the future.

Additionally, I've set up makefiles and GitHub tooling to allow automatic linting/testing based on the main OAuth2 Proxy repo.

### CODE REVIEW

**Review by elmiko — COMMENTED** [review]
i tried this out and everything worked as expected. i think it would be nice to add something to the readme letting people know how to run these tests and also what pre-reqs they might need. for example, i had to install golangci-lint. just a suggestion, otherwise lgtm

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+<!--- Please describe in detail how you tested your changes. -->
+<!--- Include details of your testing environment, and the tests you ran to -->
+<!--- see how your change affects other areas of the code, etc. -->
+
+## Checklist:
+
+<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
+<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
+
+- [ ] My change requires a change to the documentation or CHANGELOG.
```
Comment (NickMeves) [inline .github/PULL_REQUEST_TEMPLATE.md]: I recall these we're the greatest default checkboxes even over in the main project -- are these the checkboxes we want to add?

(Non-blocking).

File: reference-gen/pkg/generator/testdata/types.go [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+// aspects of the code generation.
+type MyTestStruct struct {
+	// Name is the name of the MyTestStruct.
+	Name string `json:"name"`
+
+	// privateField is a private field and so should not be documented
+	// in the generated docs.
+	privateField string
+
+	// LongMessageInt has a very long message, very very very very very very
```
Comment (NickMeves) [inline reference-gen/pkg/generator/testdata/types.go:18]: 🤣

**Review by NickMeves — APPROVED** [review]
1 non-blocking NIT on the PR template.

Otherwise, LGTM! Thanks for all the added tests!

**Comment by JoelSpeed** [comment]
@elmiko @NickMeves thanks for the suggestions! I added a readme with some brief details and a check in the makefile to check if golangci-lint is installed, I think this should address your suggestions

**Review by elmiko — COMMENTED** [review]
readme looks nice , +1

**Pull request merged**